src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -32,35 +33,54 @@ public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, o
             try
             {
                 string value = (string)data;
```

Comment:
Ah. This is another reason for the try…catch mechanism; if this throws it's also turned into the correct exception down the line. This though could be `if (!(data is string value)) throw CreateExceptionForInvalidValue();`